src/util/ivmFactory.js

```diff
@@ -243,6 +243,11 @@ async function createIvm(code, libraryVersionIds, versionId, secrets, testMode)
     }),
   );
 
+  await jail.set('_credentials', function (...args) {
+    if (args.length == 0 || !credentials || !credentials[args[0]]) return 'ERROR';
```

Comment:
Also let's try and break the condition into 3 parts right:
1. Customer provided wrong data.
2. Credentials are not present meaning in the ivm at all.
3. No lookup is available in the credentials.

Handling would be different for each one of them. We should not return generic `ERROR`.